src/Magento/FunctionalTestingFramework/DataGenerator/Persist/CurlHandler.php

```diff
@@ -95,6 +102,27 @@ public function __construct($operation, $entityObject, $storeCode = null)
         $this->isJson = false;
     }
 
+    /**
+     * Get CurlHandler instance.
+     *
+     * @param string           $operation
+     * @param EntityDataObject $entityObject
+     * @param string|null      $storeCode
+     *
+     * @return CurlHandler
+     */
+    public static function getInstance(
+        string $operation,
+        EntityDataObject $entityObject,
+        ?string $storeCode = null
+    ) {
+        if (self::$INSTANCE === null) {
+            return new self($operation, $entityObject, $storeCode);
+        }
+
+        return self::$INSTANCE;
+    }
+
```

Comment:
@anzin CurlHandler is widely used in MFTF and called repeatedly in test runs. Changing it to singleton may not work. Let me know if you think otherwise.

Reply:
Hello @jilu1, I added new changes by deleting singleton, please check if this solution is better.

Thank you.